lms/djangoapps/certificates/tests/test_services.py

```diff
@@ -61,3 +66,27 @@ def test_invalidate_certificate(self):
                 'status': CertificateStatuses.unavailable
             }
         )
+
+    @override_waffle_flag(CERTIFICATES_USE_ALLOWLIST, active=True)
+    def test_invalidate_certificate_allowlist(self):
+        """
+        Verify that CertificateService does not invalidate the certificate if it is allowlisted
+        """
+        u = UserFactory.create()
+        c = CourseFactory()
+        course_key = c.id  # pylint: disable=no-member
+        GeneratedCertificateFactory.create(
+            status=CertificateStatuses.downloadable,
+            user=u,
+            course_id=course_key,
+            grade=1.0
+        )
+        CertificateWhitelistFactory(
+            user=u,
+            course_id=course_key
+        )
+        success = self.service.invalidate_certificate(u.id, course_key)
+        self.assertFalse(success)
+
+        cert = GeneratedCertificate.objects.get(user=u.id, course_id=course_key)
+        self.assertEquals(cert.status, CertificateStatuses.downloadable)
```

Comment:
nit: I believe `assertEquals` is the deprecated alias to assertEqual: https://docs.python.org/3/library/unittest.html#deprecated-aliases. I think I remember seeing a warning last time I had used it locally.